dotnet/src/Plugins/Plugins.Core/CodeInterpreter/SessionsPythonPlugin.cs

```diff
@@ -0,0 +1,302 @@
+        {
+            Headers = {
+                { "Authorization", $"Bearer {(await this.GetAuthTokenAsync().ConfigureAwait(false))}" },
+                { "User-Agent", $"{HttpHeaderConstant.Values.UserAgent}/{s_assemblyVersion} (Language=dotnet)" }
+            },
+            Content = new MultipartFormDataContent
+            {
+                { fileContent, "file", remoteFilePath },
+            }
+        };
+
+        var response = await httpClient.SendAsync(request).ConfigureAwait(false);
+
+        if (!response.IsSuccessStatusCode)
+        {
+            throw new HttpRequestException($"Failed to upload file. Status code: {response.StatusCode}");
+        }
+
+        var JsonElementResult = JsonSerializer.Deserialize<JsonElement>(await response.Content.ReadAsStringAsync().ConfigureAwait(false));
+        // Response Body {"$id":"1","$values":[{"$id":"2","filename":"test-file.txt","size":17,"last_modified_time":"2024-05-07T22:24:00.7528208Z"}]}
+
+        return JsonSerializer.Deserialize<SessionRemoteFileMetadata>(JsonElementResult.GetProperty("$values")[0].GetRawText())!;
+    }
+
+    /// <summary>
+    /// Downloads a file from the current Session ID.
+    /// </summary>
+    /// <param name="remoteFilePath"> The path to download the file from, relative to `/mnt/data`. </param>
+    /// <param name="localFilePath"> The path to save the downloaded file to. If not provided, the file is returned as a BufferedReader. </param>
+    /// <returns> The data of the downloaded file. </returns>
+    [Description("Downloads a file from the current Session ID.")]
+    public async Task<byte[]?> DownloadFileAsync(
+        [Description("The path to download the file from, relative to `/mnt/data`.")] string remoteFilePath,
+        [Description("The path to save the downloaded file to. If not provided, the file is returned as a BufferedReader.")] string localFilePath)
+    {
+        Verify.NotNullOrWhiteSpace(remoteFilePath, nameof(remoteFilePath));
+
+        if (this._logger.IsEnabled(LogLevel.Information))
+        {
+            this._logger.LogInformation("Downloading file: {RemoteFilePath} to {LocalFilePath}", remoteFilePath, localFilePath);
+        }
+
+        using var httpClient = this._httpClientFactory.CreateClient();
+        httpClient.DefaultRequestHeaders.Add("Authorization", $"Bearer {(await this.GetAuthTokenAsync().ConfigureAwait(false))}");
+        httpClient.DefaultRequestHeaders.Add("User-Agent", $"{HttpHeaderConstant.Values.UserAgent}/{s_assemblyVersion} (Language=dotnet)");
+
+        var response = await httpClient.GetAsync($"{this._poolManagementEndpoint}python/downloadFile?identifier={this._settings.SessionId}&filename={remoteFilePath}").ConfigureAwait(false);
+        if (!response.IsSuccessStatusCode)
+        {
+            throw new HttpRequestException($"Failed to download file. Status code: {response.StatusCode}");
```

Comment:
Include response body (if any) as it may contain additional diagnostic information